src/test/java/seedu/loyaltylift/logic/commands/AdvanceOrderStatusCommandTest.java

```diff
@@ -0,0 +1,74 @@
+package seedu.loyaltylift.logic.commands;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.loyaltylift.logic.commands.AdvanceOrderStatusCommand.MESSAGE_ADVANCE_STATUS_SUCCESS;
+import static seedu.loyaltylift.logic.commands.CommandResult.ListViewGuiAction.LIST_AND_SHOW_ORDER;
+import static seedu.loyaltylift.logic.commands.CommandTestUtil.assertCommandFailure;
+import static seedu.loyaltylift.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static seedu.loyaltylift.model.order.StatusUpdate.DATE_FORMATTER;
+import static seedu.loyaltylift.testutil.TypicalAddressBook.getTypicalAddressBook;
+import static seedu.loyaltylift.testutil.TypicalIndexes.INDEX_FIRST;
+import static seedu.loyaltylift.testutil.TypicalIndexes.INDEX_SECOND;
+
+import java.time.LocalDate;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.loyaltylift.commons.core.Messages;
+import seedu.loyaltylift.commons.core.index.Index;
+import seedu.loyaltylift.model.AddressBook;
+import seedu.loyaltylift.model.Model;
+import seedu.loyaltylift.model.ModelManager;
+import seedu.loyaltylift.model.UserPrefs;
+import seedu.loyaltylift.model.order.Order;
+import seedu.loyaltylift.testutil.OrderBuilder;
+
+public class AdvanceOrderStatusCommandTest {
+    private Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+
+    @Test
+    public void execute_invalidIndexUnfilteredList_throwsCommandException() {
+        Index outOfBoundIndex = Index.fromOneBased(model.getFilteredOrderList().size() + 1);
+        AdvanceOrderStatusCommand advanceOrderStatusCommand = new AdvanceOrderStatusCommand(outOfBoundIndex);
+
+        assertCommandFailure(advanceOrderStatusCommand, model, Messages.MESSAGE_INVALID_ORDER_DISPLAYED_INDEX);
+    }
+
+    @Test
+    public void execute_validIndexUnfilteredList_success() {
+        LocalDate dateToday = LocalDate.now();
+        String formattedDate = dateToday.format(DATE_FORMATTER);
+        Order orderToAdvance = model.getFilteredOrderList().get(INDEX_FIRST.getZeroBased());
+        AdvanceOrderStatusCommand advanceOrderStatusCommand = new AdvanceOrderStatusCommand(INDEX_FIRST);
+
+        Order advancedOrder = new OrderBuilder(orderToAdvance).withNextStatus(formattedDate).build();
+        String expectedMessage = String.format(MESSAGE_ADVANCE_STATUS_SUCCESS,
+                advancedOrder, advancedOrder.getStatus().getLatestStatus());
+        Model expectedModel = new ModelManager(new AddressBook(model.getAddressBook()), new UserPrefs());
+        expectedModel.setOrder(model.getFilteredOrderList().get(0), advancedOrder);
+        CommandResult expectedCommandResult = new CommandResult(expectedMessage, LIST_AND_SHOW_ORDER);
+        assertCommandSuccess(advanceOrderStatusCommand, model, expectedCommandResult, expectedModel);
+    }
+
```

Comment:
We should test for command failure if the command is already completed or cancelled